src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -771,7 +826,7 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
 
                 var invocationRequest = await context.ToRpcInvocationRequest(_workerChannelLogger, _workerCapabilities, _isSharedMemoryDataTransferEnabled, _sharedMemoryManager);
                 AddAdditionalTraceContext(invocationRequest.TraceContext.Attributes, context);
-                _executingInvocations.TryAdd(invocationRequest.InvocationId, context);
+                _executingInvocations.TryAdd(invocationRequest.InvocationId, new(context, _messageDispatcherFactory.Create(invocationRequest.InvocationId)));
```

Comment:
So when the feature is enabled we create a new instance of `OrderedInvocationMessageDispatcher` for each invocation and when not enabled we use the same instance of `ThreadPoolInvocationProcessorFactory` for all invocation (because the implementation is straight forward - queuing to TP)?

Reply:
never mind. I missed the comment in `ThreadPoolInvocationProcessorFactory`

Reply:
Yeah, that's correct... we don't really need per-invocation state stored with the ThreadPool invocation. I really was trying to avoid adding these factories/types for the current scenario, but it was *very* messy without them. I think this cleans it up and shouldn't add overhead to how things currently run while we get confidence in the new mode.